feat: add default image when no image is found

GabrielPrediger opened the pull request:
The main idea was to try to add the possibility of informing a default image when no token is found, as there are many tokens in testnet that do not have an image and sometimes mainnet does not either, so it allows us the chance to make our dapps more beautiful in these cases.

Please, you are more than welcome to adjust my code with your commits, I confess that I got a little lost in the code so I possibly didn't create this feature in the best way

0xa3k5 commented:
awesome @GabrielPrediger ngl this had come to my mind and its a good idea to make it a prop actually, i was thinking about rendering the Web3Icon's logo as a fallback. I will review the PR and probably extend it to work with other icon components, WalletIcon, NetworkIcon.. 

thanks!

0xa3k5 commented:
hey @GabrielPrediger I implemented the necessary changes and refactored the code a little bit. you can try using it with passing a `fallback` prop to the `TokenIcon` `WalletIcon` or `NetworkIcon`. I also made the `fallback` accept and handle either a url (string) or a ReactNode so if you are importing a image as a React component you can pass that as well.

Thank you for your contribution

GabrielPrediger commented:
> hey @GabrielPrediger I implemented the necessary changes and refactored the code a little bit. you can try using it with passing a `fallback` prop to the `TokenIcon` `WalletIcon` or `NetworkIcon`. I also made the `fallback` accept and handle either a url (string) or a ReactNode so if you are importing a image as a React component you can pass that as well.
> 
> Thank you for your contribution

you are wellcome mate, congrats for you lib, it's dope
